app/services/articles/attributes.rb

```diff
@@ -11,14 +11,17 @@ def initialize(attributes, article_user)
       @article_user = article_user
     end
 
-    def for_update(update_edited_at: false, update_published_at: false)
+    def for_update(update_edited_at: false, update_published_at: false,
+                   nullify_published_at: false)
```

Comment:
This is probably a minor nit, but: the method signature allows `update_published_at` and `nullify_published_at` to both be set, however, the `if/elsif` means that only one can ever be active (it appears that `update` takes first priority).

Maybe a guard clause solves the surface issue (options that are sort of exclusive) but it also seems like there's extra confusion or complexity here, where `Updater` has some responsibility for determining a mode `(un)publishing?` but `Attributes` receives that information in the form of these options, which it then has to recompose into `published_at`.

Probably we already looked at this and there are other reasons this won't work, but another approach might be to override the `published` setter (or proxy it in a virtual attribute, `is_(un)published` or something) that writes into the `published_at` as needed.